### PR TITLE
Fix AddJsonFilenameFlag function

### PR DESCRIPTION
**Pull request by tcharding**
**What this PR does / why we need it**:

Currently `AddJsonFilenameFlag` is defined in `bash-comp-utils.go`. This is the only definition within this file. The function is only called from within `cmd/*.go` so its proper place is in that package.

The name of the function is incorrect (Json vs JSON notwithstanding). The `--filename` flag accepts multiple filename extensions, the name is therefore misleading. We should rename it `AddFilenameFlag`.

This PR renames the function and moves the definition into `cmd/utils/helpers.go` along with other `Add*Flag` functions. `bash-comp-utils.go` is removed from the tree.

**Special notes for your reviewer**:

This PR highlights what may be a missing feature (or a bug). Most `kubectl` commands offer the `--recursive` flag when using `--filename`. Should this be the case for all commands? Commands not offering the `--recursive` flag are
```
pkg/kubectl/cmd/apply_set_last_applied.go
pkg/kubectl/cmd/rollingupdate.go
pkg/kubectl/cmd/set/set_resources.go
```
Perhaps `AddFilenameFlag` should not exist at all?

**Release note**:

```release-note
NONE
```
/sig cli
/kind cleanup

### CODE REVIEW

File: pkg/kubectl/bash_comp_utils.go [hunk 1]
```diff
@@ -1,36 +0,0 @@
-/*
```
Comment (xiangpengzhao) [inline pkg/kubectl/bash_comp_utils.go]: You may need to run `hack/update-bazel.sh` after removing this file.

**Comment by tcharding** [comment]
thank @xiangpengzhao bazel-updated!

**Comment by rootfs** [comment]
@tcharding squash into one commit

File: pkg/kubectl/cmd/set/set_resources.go [hunk 2]
```diff
@@ -111,7 +111,7 @@ func NewCmdResources(f cmdutil.Factory, out io.Writer, errOut io.Writer) *cobra.
 
 	cmdutil.AddPrinterFlags(cmd)
 	//usage := "Filename, directory, or URL to a file identifying the resource to get from the server"
-	//kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, usage)
+	//kubectl.cmdutil.AddFilenameFlag(cmd, &options.Filenames, usage)
```
Comment (xiangpengzhao) [inline pkg/kubectl/cmd/set/set_resources.go]: nit: do we still want to keep the commented lines?

Comment (tcharding) [reply pkg/kubectl/cmd/set/set_resources.go]: Well spotted. Thanks.

**Pull request closed without merging**